ecs-cli/modules/utils/compose/ecs_params_reader.go

```diff
@@ -25,8 +27,10 @@ import (
 type ECSParams struct {
 	Version        string
 	TaskDefinition EcsTaskDef `yaml:"task_definition"`
+	RunParams      RunParams  `yaml:"run_params"`
 }
 
+// ECS TaskDefinition fields
 type EcsTaskDef struct {
 	NetworkMode          string        `yaml:"ecs_network_mode"`
```

Comment:
This is probably showing my unfamiliarity with this code base. But, where do we validate that `NetworkMode` is set to a valid/supported network mode in ECS?

Reply:
We don't validate it in the CLI -- we bubble up the error returned by ECS. Example:

```
~/sandbox/test_fields$ cat ecs-params.yml.tn 
version: 1
task_definition:
  ecs_network_mode:foo 
run_params:
  network_configuration:
    awsvpc_configuration:
      subnets: [subnet-c0ffeefe]

~/sandbox/test_fields$ vim !$
vim ecs-params.yml.tn

~/sandbox/test_fields$ ecs-cli compose  --ecs-params ecs-params.yml.tn --cluster test-tn up
WARN[0000] Skipping unsupported YAML option...           option name=networks
WARN[0000] Skipping unsupported YAML option for service...  option name=networks service name=mysql
WARN[0000] Skipping unsupported YAML option for service...  option name=networks service name=wordpress
ERRO[0000] Error registering task definition             error="ClientException: foo is not a valid network mode. Must be one of [bridge,host,none]\n\tstatus code: 400, request id: <my request id>" family="test_fields"
ERRO[0000] Create task definition failed                 error="ClientException: foo is not a valid network mode. Must be one of [bridge,host,none]\n\tstatus code: 400, request id: <my request id>"
FATA[0000] ClientException: foo is not a valid network mode. Must be one of [bridge,host,none]

```

Reply:
alright. `"Must be one of [bridge,host,none]"` seems like we have to improve that error message to include `"awsvpc"` (I know its outside this PRs scope, just making a note for self).

Reply:
Yeah, I already talked to control plane about this and there's a fix being pushed out later this week.